src/action.ts

```diff
@@ -2,55 +2,75 @@ import * as core from '@actions/core'
 import * as util from 'util'
 import * as child_process from 'child_process'
 
-const { GITHUB_ACTOR } = process.env
+class GitArgs {
+  readonly command: string
+
+  constructor(
+    readonly ref: string,
+    readonly directory: string,
+  ) {
+    this.command = `git -C ${this.directory}`
+  }
+}
 
 async function exec(command: string) {
   const { stdout, stderr } = await util.promisify(child_process.exec)(command)
   if (stderr) console.error(stderr)
   return stdout
 }
 
-function annotatedTag(message: string, ref: string) {
+function annotatedTag(message: string, git: GitArgs) {
   core.info('Creating annotated tag...')
-  return exec(`git tag -a -f -m "${message}" ${ref}`)
+  return exec(`${git.command} tag -a -f -m "${message}" ${git.ref}`)
 }
 
-function lightweightTag(ref: string) {
+function lightweightTag(git: GitArgs) {
   core.info('Creating lightweight tag...')
-  return exec(`git tag -f ${ref}`)
+  return exec(`${git.command} tag -f ${git.ref}`)
 }
 
-function forceBranch(ref: string) {
+function forceBranch(git: GitArgs) {
   core.info('Updating branch...')
-  return exec(`git branch -f ${ref}`)
+  return exec(`${git.command} branch -f ${git.ref}`)
 }
 
-async function run() {
-  try {
+function setupUser(git: GitArgs) {
```

Comment:
This should be an async function, right?

```suggestion
async function setupUser(git: GitArgs) {
```

Reply:
Oh, yes. Thanks for the catch!